src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -2037,7 +2011,15 @@ public async Task<JObject> CreateJObjectForVariableValue(MonoBinaryReader retDeb
                 case ElementType.Class:
                 case ElementType.Object:
                 {
-                    ret = await CreateJObjectForObject(retDebuggerCmdReader, typeIdFromAttribute, forDebuggerDisplayAttribute, token);
+                    var objectId = retDebuggerCmdReader.ReadInt32();//has to stay here
+                    var isDelegate = await IsDelegate(objectId, token);
+                    var (className, description) = await GetObjectClassNameAndDescription(objectId, isDelegate);
+                    if (isDelegate && description == "")
+                    {
+                        ret = CreateJObject(className.ToString(), "symbol", className.ToString(), false);
+                        break;
+                    }
+                    ret = CreateJObjectForObject(className.ToString(), description, objectId);
```

Comment:
What do you mean by the functional change? It is just moving the code that used to be in https://github.com/dotnet/runtime/blob/eb51b02b158c3ff71a1ec7eac8a211d1d464c1a5/src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs#L1821 with some small refactoring (saving IsDelegate in a variable and reorganizing the logic) so that we can extract a simplified, re-usable version of CreateJObjectForObject method, without MonoReader.